src/components/controls/TextArea/TextAreaControl.tsx

```diff
@@ -2,78 +2,69 @@ import React from 'react';
         name,
         id,
         tabIndex,
-        autoFocus,
         autoComplete,
         placeholder,
         value,
         defaultValue,
+        controlRef,
+        controlProps,
+        size,
+        rows,
+        minRows = 1,
+        maxRows,
+        autoFocus,
+        disabled,
         onChange,
         onFocus,
         onBlur,
         onKeyDown,
         onKeyUp,
         onKeyPress,
-        controlRef,
-        controlProps,
-        disabled,
-        rows,
-        minRows = 1,
-        maxRows,
     } = props;
     const innerControlRef = React.useRef<HTMLTextAreaElement>(null);
     const handleRef = useForkRef(controlRef, innerControlRef);
     const textareaRows = rows || minRows;
+    const innerValue = value || innerControlRef?.current?.value;
 
     const resizeHeight = React.useCallback(() => {
         const control = innerControlRef?.current;
 
         if (control && !rows) {
-            const inputValue = value || control.value;
-            const numberOfLines = (inputValue.match(/\n/g) || []).length + 1;
-
             const controlStyles = getComputedStyle(control);
-
             const lineHeight = parseInt(controlStyles.getPropertyValue('line-height'), 10);
-            const borderWidth = parseInt(controlStyles.getPropertyValue('border-top-width'), 10);
             const paddingTop = parseInt(controlStyles.getPropertyValue('padding-top'), 10);
+            const linesWithCarriageReturn = (innerValue?.match(/\n/g) || []).length + 1;
+            const linesByScrollHeight = Math.floor(control.scrollHeight / lineHeight);
 
-            const lines = Math.floor(control.scrollHeight / lineHeight);
-
-            if (maxRows && maxRows < Math.max(lines, numberOfLines)) {
+            if (maxRows && maxRows < Math.max(linesByScrollHeight, linesWithCarriageReturn)) {
                 control.style.height = 'auto';
-                control.style.height = `${
-                    maxRows * lineHeight + 2 * paddingTop + 2 * borderWidth
-                }px`;
+                control.style.height = `${maxRows * lineHeight + 2 * paddingTop}px`;
             } else {
                 control.style.height = 'auto';
-                control.style.height = `${control.scrollHeight + 2 * borderWidth}px`;
+                control.style.height =
+                    linesWithCarriageReturn > 1 || linesByScrollHeight > 1
+                        ? `${control.scrollHeight}px`
+                        : ''; // Set falsy value to use styles from css
             }
         }
-    }, [rows, maxRows, value]);
+    }, [rows, maxRows, innerValue]);
 
-    React.useEffect(resizeHeight, [resizeHeight]);
-
-    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>): void => {
-        if (onChange) {
-            onChange(event);
-        } else {
-            /** In case uncontrolled component */
-            resizeHeight();
-        }
-    };
+    React.useEffect(resizeHeight, [resizeHeight, size]);
```

Comment:
I suggest you to write full form:
```js
React.useEffect(() => {
    resizeHeight();
}, [resizeHeight, size])
```
Hook function has a specific signature and expects function in result, better not to mix